app/Models/Image.php

```diff
@@ -154,6 +125,37 @@ public static function upload(UploadedFile $file, $user, bool $forceReupload = f
         }
     }
 
+    /**
+     * Converts the image into the specified cover type to the specified path.
+     *
+     * @param File $image The image file to be processed
+     * @param string $path The path to save the processed image file
+     * @param int $coverType The type to process the image to
+     */
+    public static function processFile(File $image, string $path, $coverType) {
```

Comment:
Like I said in another comment above, this should be possible to refactor into a public instance method (a wrapper method that rebuilds all covers) and a private one (that rebuilds a single cover, and will likely contain most of this function's current form). Aim to contain any knowledge about how the files are stored on disk within this class - you shouldn't need to expose the file path for most purposes in PFM.

Reply:
Nice refactoring. 🙂 This function should be made private now unless it's being used anywhere outside of this file.